### DISCUSSION TITLE
Livereload with tailwind 3

### DISCUSSION THREAD

**Discussion by romanoff** [tailwindlabs/tailwindcss#6416] [Help] [answered]
Was following tailwind 3 tutorial here: https://tailwindcss.com/docs/installation and tried to setup livereload (have seen that approach in this egghead tutorial: https://egghead.io/courses/introduction-to-tailwind-and-the-utility-first-workflow-ac67 . The problem that I am facing is that even though css is automatically reloaded, html is not. If I turn off auto- generation of css then html is reloaded. Not sure what the issue is. Also would be great if live- reload setup was described somewhere in docs (as example) 

[tailwind_livereload_problem.zip](https://github.com/tailwindlabs/tailwindcss/files/7698619/tailwind_livereload_problem.zip)

**Comment by NovusOpus** [comment, marked as answer]
This is the Official [Tailwind CSS Youtube Channel](https://www.youtube.com/c/TailwindLabs/videos) You'll find the latest Tutorials and and videos. Since the Course you mentioned was intended for tailwind V1, And was updated 10 months ago, A lot has changed in Tailwind since then, so you should check the Oficial Tailwind CSS Youtube Channel for more recent Videos and Tutorials.

This video from the Channel show you a tutorial for: [Setting Up Tailwind](https://youtu.be/DxcJbrs6rKk?t=14). It's from version V2.2 of Tailwind but works with V3 making a few changes from the tutorial:

1. You don't have to use `--jit` anymore since it's already enabled by default in the V3
2. And instead of `--purge`, we now use `--content`.

 So with the changes it should look like this:
```console
npx tailwindcss -o ./build/tailwind.css --content "./**/*.html" -w
```

Instead of:
```console
npx tailwindcss -o ./build/tailwind.css --jit --purge"./**/*.html" -w
```

Apart from those 2 changes you can follow anything else as showed in the `All-new Tailwind CLI tool` section of video.

Another thing, If you using Visual Studio Code a nice extension for Live Reload is: [Five Server (Live Server)](https://marketplace.visualstudio.com/items?itemName=yandeu.five-server)

**Reply by romanoff** [reply]
Thank you for your comment. Will look more into those screencasts. One thing though is that even after changing command to watch for changes to `npx tailwindcss -o ./build/tailwind.css --jit --purge"./**/*.html" -w` instead of the previous one, I still have the same problem I had before. The problem is that I need to manually reload html. Otherwise html changes are not picked up.

Here is how to reproduce issue:
1. Get zip file [tailwind_livereload_problem.zip](https://github.com/tailwindlabs/tailwindcss/files/7701669/tailwind_livereload_problem.zip)
2. Unzip it
3. npm install -g live-server
4. Run `make server` inside the folder
5. Go to http://127.0.0.1:8080/src/
Now livereload works (except css). Changes of html are picked up. Great
6. Start `make watch` 
After this looks like css changes are picked up, but now html isn't 

Not sure why that is. Would be great if somebody could look into it. Alternatively would be happy to see some other example of minimal project (having one simple index.html html page) of tailwind 3 setup with livereload. Thanks a lot

**Reply by NovusOpus** [reply]
@romanoff Oh I think the problem is with live-server(Last update was 3 years ago). 

Try using the one I recommended Five-Server, It's a maintained and updated fork of Live Server, Works great with tailwind. Use it like this: 

Install Five-Server globally with: 
`npm -g i five-server`

Then run this in your project directory(from where you're running `make server`): 
`five-server`

Go to: 
[localhost:5500/src/](http://localhost:5500/src/)

You can read more about further setup(Changing default port, directories and other stuff, This is Optional) in the Five-Server [Npm Page](https://www.npmjs.com/package/five-server)

**Reply by romanoff** [reply]
Thanks a lot. This worked :)
